pkg/task/task.go

```diff
@@ -146,3 +207,52 @@ type CheckpointInfo struct {
 type CompletedTask struct {
 	Checkpoint TaskCheckpoint `json:"checkpoint"`
 }
+
+func newCheckpointInfo() CheckpointInfo {
+	return CheckpointInfo{
+		CompletedTasks: make(map[string]*CompletedTask),
+	}
+}
+
+func (c CheckpointInfo) taskCompleted(name string, completedTask *CompletedTask) {
+	c.CompletedTasks[name] = completedTask
+}
+
+func GetCheckpointFile(file string) (*CheckpointInfo, error) {
+	logger.V(4).Info("Reading checkpoint", "file", file)
+	content, err := os.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading checkpoint file: %v\n", err)
+	}
+	checkpointInfo := &CheckpointInfo{}
+	err = yaml.Unmarshal(content, checkpointInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshalling checkpoint: %v\n", err)
+	}
+
+	return checkpointInfo, nil
+}
+
+func UnmarshalTaskCheckpoint(taskCheckpoint TaskCheckpoint, config interface{}) error {
```

Comment:
Can we take checkpoint type instead of `interface{}`?

Reply:
We are passing in the checkpoint type as the second parameter when we actual call this method, so it unmarshals into that type. We just have it as `interface{} `type in the function definition so that we can accept whatever checkpoint type was needed for the task

Reply:
I was suggesting this
```suggestion
func UnmarshalTaskCheckpoint(taskCheckpoint TaskCheckpoint, config TaskCheckpoint) error {
```